clang/include/clang/CIR/Dialect/IR/CIRAttrs.td

```diff
@@ -220,6 +220,27 @@ def IntAttr : CIR_Attr<"Int", "int", [TypedAttrInterface]> {
   let hasCustomAssemblyFormat = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// PointerAttr
+//===----------------------------------------------------------------------===//
+
+def PtrAttr : CIR_Attr<"Ptr", "ptr", [TypedAttrInterface]> {
```

Comment:
I like where this is going, it's time to generalize pointer constants, `NullAttr` is in fact a specialization of this attribute. The name should be more descriptive, something like `ConstPtrAttr` and we should get rid of `NullAttr`, given that is covered by `ConstPtrAttr::isNullValue()`. Can you do this change in a another PR that this should depend on?

Still going through the PR, but wanted to antecipate this one.

Reply:
Sounds good to make it a separate change. 

When it comes to replacing `NullAttr` with `ConstPtrAttr`, I'm afraid that the current printing, i.e, `cir.null` may not be honored, as IIUC the attribute name will always be printed, even with a custom printer. Do you think that's ok? But correct me if I'm wrong.

Reply:
I reviewed the other PR before reading this, it's a good point, I'm all in for something along the lines of `cir.ptr<null>`